pyretailscience/plots/histogram.py

```diff
@@ -0,0 +1,256 @@
+    range_upper: float | None,
+    range_method: Literal["clip", "fillna"],
+) -> pd.DataFrame:
+    """Applies range clipping or filling based on the provided method and returns the modified dataframe.
+
+    Args:
+        df (pd.DataFrame): The dataframe to apply range clipping to.
+        value_col (list of str): The column(s) to apply clipping or filling to.
+        range_lower (float, optional): Lower bound for clipping or filling NA values.
+        range_upper (float, optional): Upper bound for clipping or filling NA values.
+        range_method (Literal, optional): Whether to "clip" values outside the range or "fillna". Defaults to "clip".
+
+    Returns:
+        pd.DataFrame: The modified dataframe with the clipping or filling applied.
+    """
+    if range_lower is not None or range_upper is not None:
+        if range_method == "clip":
+            return df.assign(**{col: df[col].clip(lower=range_lower, upper=range_upper) for col in value_col})
+        return df.assign(
+            **{
+                col: df[col].apply(
+                    lambda x: np.nan
+                    if (range_lower is not None and x < range_lower) or (range_upper is not None and x > range_upper)
+                    else x,
+                )
+                for col in value_col
+            },
+        )
+
+    return df
+
+
+def _get_num_histograms(df: pd.DataFrame, value_col: list[str], group_col: str | None) -> int:
+    """Calculates the number of histograms to be plotted.
+
+    Args:
+        df (pd.DataFrame): The dataframe being plotted.
+        value_col (list of str): The column(s) being plotted.
+        group_col (str, optional): The column used for grouping data into histograms.
+
+    Returns:
+        int: The number of histograms to plot.
+    """
+    num_histograms = len(value_col)
+
+    if group_col is not None:
+        num_histograms = max(num_histograms, df[group_col].nunique())
+
+    return num_histograms
+
+
+def _plot_histogram(
+    df: pd.DataFrame,
+    value_col: list[str],
+    group_col: str | None,
+    ax: Axes | None,
+    cmap: ListedColormap,
+    num_histograms: int,
+    **kwargs: dict,
+) -> Axes:
+    """Plots histograms for the provided dataframe.
+
+    Args:
+        df (pd.DataFrame): The dataframe to plot.
+        value_col (list of str): The column(s) to plot.
+        group_col (str, optional): The column used to group data into multiple histograms.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        cmap (ListedColormap): The colormap to use for the plot.
+        num_histograms (int): The number of histograms being plotted.
+        **kwargs: Additional keyword arguments for Pandas' `plot` function.
+
+    Returns:
+        Axes: The matplotlib axes object with the plotted histogram.
+    """
+    add_legend = num_histograms > 1
+
+    if group_col is None:
+        return df[value_col].plot(
+            kind="hist",
+            ax=ax,
+            alpha=0.5,
```

Comment:
Can you make it so that you check kwargs for a key with alpha in it and use that, otherwise use a default value for alpha. Basically, make it so alpha can get overridden.

Also do you think alpha 0.5 is too transparent?